Update Kconfig

TheFeralEngineer opened the pull request:
add 48KiB bootloader option to STM32F401 for Artillery Sidewinder X3 boards

JamesH1978 commented:
Thank you for submitting a PR. You will need to fix your code remove the trading spaces as per the checks error, And you will need to also need to do the regression tests and sign off your PR. Point 4 in the guidelines in https://github.com/Klipper3d/klipper/blob/master/docs/Example_Configs.md and point 3 in What to expect in https://github.com/Klipper3d/klipper/blob/master/docs/CONTRIBUTING.md

Thanks
James

TheFeralEngineer commented on src/stm32/Kconfig:
```diff
@@ -288,7 +288,7 @@ choice
         bool "36KiB bootloader" if MACH_STM32F1
     config STM32_FLASH_START_C000
-        bool "48KiB bootloader" if MACH_STM32F4x5
+        bool "48KiB bootloader" if MACH_STM32F4x5 || MACH_STM32F401
```
add stm32f401 to 48KiB for Artillery Sidewinder X3

TheFeralEngineer reviewed (commented):
added additional information

TheFeralEngineer reviewed (commented):
removed whitespaces

TheFeralEngineer commented:
> Thank you for submitting a PR. You will need to fix your code remove the trading spaces as per the checks error, And you will need to also need to do the regression tests and sign off your PR. Point 4 in the guidelines in https://github.com/Klipper3d/klipper/blob/master/docs/Example_Configs.md and point 3 in What to expect in https://github.com/Klipper3d/klipper/blob/master/docs/CONTRIBUTING.md
> 
> Thanks James

I'm completely new to this PR stuff, so bear with me...

TheFeralEngineer reviewed (commented):
basic .cfg

TheFeralEngineer reviewed (commented):
removed another whitespace

TheFeralEngineer reviewed (commented):
removed uppercase letters in config name

JamesH1978 commented:
Thank you for the changes you have made so far but you still need to provide a [test/klippy/printers.test](https://github.com/Klipper3d/klipper/blob/master/test/klippy/printers.test) file with the entry added for your new file, so it can pass regression tests as per the doc i linked.

Thanks
James

Sineos commented on config/printer-artillery-sidewinder-x3-plus-2024.cfg:
```diff
@@ -0,0 +1,237 @@
+y_offset:11.6
+speed:5
+lift_speed:15
+#samples:
+#sample_retract_dist:
+#samples_result:
+#samples_tolerance:
+#samples_tolerance_retries:
```
remove or add values deviating from the defaults where necessary.

Sineos commented on config/printer-artillery-sidewinder-x3-plus-2024.cfg:
```diff
@@ -0,0 +1,237 @@
+probe_count: 10, 10
+fade_start: 1.0
+fade_end: 0.0
+#fade_target:
+#   The z position in which fade should converge. When this value is
+#   set to a non-zero value it must be within the range of z-values in
+#   the mesh. Users that wish to converge to the z homing position
+#   should set this to 0. Default is the average z value of the mesh.
+#split_delta_z: .025
+#   The amount of Z difference (in mm) along a move that will trigger
+#   a split. Default is .025.
+#move_check_distance: 5.0
+#   The distance (in mm) along a move to check for split_delta_z.
+#   This is also the minimum length that a move can be split. Default
+#   is 5.0.
+#mesh_pps: 2, 2
+#   A comma separated pair of integers X, Y defining the number of
+#   points per segment to interpolate in the mesh along each axis. A
+#   "segment" can be defined as the space between each probed point.
+#   The user may enter a single value which will be applied to both
+#   axes. Default is 2, 2.
+algorithm: bicubic
+#   The interpolation algorithm to use. May be either "lagrange" or
+#   "bicubic". This option will not affect 3x3 grids, which are forced
+#   to use lagrange sampling. Default is lagrange.
+#bicubic_tension: .2
+#   When using the bicubic algorithm the tension parameter above may
+#   be applied to change the amount of slope interpolated. Larger
+#   numbers will increase the amount of slope, which results in more
+#   curvature in the mesh. Default is .2.
+#relative_reference_index:
+#   A point index in the mesh to reference all z values to. Enabling
+#   this parameter produces a mesh relative to the probed z position
+#   at the provided index.
+#faulty_region_1_min:
+#faulty_region_1_max:
+#   Optional points that define a faulty region.  See docs/Bed_Mesh.md
+#   for details on faulty regions.  Up to 99 faulty regions may be added.
+#   By default no faulty regions are set.
```
see above

TheFeralEngineer replied:
removed

Sineos reviewed (changes requested):
Thanks for your contribution.
Some comments added and apparently a signed-off line is also missing

TheFeralEngineer commented:
/src/stm32/Kconfig: Added MACH_STM32F401 to 48KiB bootloader for Artillery Sidewinder x3 printers
/config: Added printer.cfg

all I'm trying to do is upload a printer.cfg and a small change to Kconfig. I haven't changed anything else in the source code and all of the tests I've ran have passed. I'm kinda at a loss here and don't know what else to do. 


Signed-off-by: Phil Timpson<theferalengineer@gmail.com>

Sineos commented:
Thanks for your changes.
What is missing is adding your new config to the regression testing. See https://github.com/Klipper3d/klipper/pull/6119/files for an example.

I apparently missed one thing during my review:

```
[force_move]
enable_force_move: True
```

Please do not activate such functions in the default config unless there is a compelling reason to do so.

TheFeralEngineer commented:
> Thanks for your changes. What is missing is adding your new config to the regression testing. See https://github.com/Klipper3d/klipper/pull/6119/files for an example.
> 
> I apparently missed one thing during my review:
> 
> ```
> [force_move]
> enable_force_move: True
> ```
> 
> Please do not activate such functions in the default config unless there is a compelling reason to do so.

30th time's a charm 🤞🤞

TheFeralEngineer commented:
now it keeps failing. I'm no good at this 😭

TheFeralEngineer reviewed (commented):
spelling error

JamesH1978 commented on config/printer-artillery-sidewinder-x3-plus-2024.cfg:
```diff
@@ -0,0 +1,188 @@
+rotation_distance: 17.75
+# Calibrate PID: https://www.klipper3d.org/Config_checks.html#calibrate-pid-settings
+#  - Example: PID_CALIBRATE HEATER=extruder TARGET=200
+control = pid
```
Please can you normalise these to : instead of equals

Thanks
James

JamesH1978 commented on config/printer-artillery-sidewinder-x3-plus-2024.cfg:
```diff
@@ -0,0 +1,188 @@
+min_temp: 0
+# Calibrate PID: https://www.klipper3d.org/Config_checks.html#calibrate-pid-settings
+#  - Example: PID_CALIBRATE HEATER=heater_bed TARGET=60
+control = pid
```
Same as above

JamesH1978 commented on config/printer-artillery-sidewinder-x3-plus-2024.cfg:
```diff
@@ -0,0 +1,188 @@
+y_offset:11.6
+speed:5
+lift_speed:15
+z_offset = 2.350
```
Same as above

JamesH1978 reviewed (changes requested):
Sorry to give you one more set of adjustments :) but apart from my comments it all looks good to me as well.

and good work!

Thanks
James

TheFeralEngineer commented:
oh, that's easy. I copied them directly from the bottom of my config (i guess the save_config puts them that way). I was starting to get a little distraught at all of the build failures before, but I think I've got it figured out now 🙂. I'll take care of it right away.
________________________________
From: JamesH1978 ***@***.***>
Sent: Wednesday, March 27, 2024 8:30 PM
To: Klipper3d/klipper ***@***.***>
Cc: TheFeralEngineer ***@***.***>; Author ***@***.***>
Subject: Re: [Klipper3d/klipper] Update Kconfig (PR #6534)


@JamesH1978 requested changes on this pull request.

Sorry to give you one more set of adjustments :) but apart from my comments it all looks good to me as well.

and good work!

Thanks
James

________________________________

In config/printer-artillery-sidewinder-x3-plus-2024.cfg<https://github.com/Klipper3d/klipper/pull/6534#discussion_r1542178413>:

> +dir_pin: !PC10
+enable_pin: !PC12
+microsteps: 64
+nozzle_diameter: 0.400
+filament_diameter: 1.750
+heater_pin: PA6
+sensor_type: EPCOS 100K B57560G104F #Generic 3950
+sensor_pin: PC5
+min_extrude_temp: 170
+min_temp: 0
+max_temp: 300
+# Calibrate E-Steps https://www.klipper3d.org/Rotation_Distance.html#calibrating-rotation_distance-on-extruders
+rotation_distance: 17.75
+# Calibrate PID: https://www.klipper3d.org/Config_checks.html#calibrate-pid-settings
+#  - Example: PID_CALIBRATE HEATER=extruder TARGET=200
+control = pid


Please can you normalise these to : instead of equals

Thanks
James

________________________________

In config/printer-artillery-sidewinder-x3-plus-2024.cfg<https://github.com/Klipper3d/klipper/pull/6534#discussion_r1542178606>:

> +#  - Example: PID_CALIBRATE HEATER=extruder TARGET=200
+control = pid
+pid_kp = 30.356
+pid_ki = 1.857
+pid_kd = 124.081
+# Calibrate PA: https://www.klipper3d.org/Pressure_Advance.html
+
+[heater_bed]
+heater_pin: PA7
+sensor_type: EPCOS 100K B57560G104F
+sensor_pin: PC0
+max_temp: 100
+min_temp: 0
+# Calibrate PID: https://www.klipper3d.org/Config_checks.html#calibrate-pid-settings
+#  - Example: PID_CALIBRATE HEATER=heater_bed TARGET=60
+control = pid


Same as above

________________________________

In config/printer-artillery-sidewinder-x3-plus-2024.cfg<https://github.com/Klipper3d/klipper/pull/6534#discussion_r1542178695>:

> +off_below: 0.19
+max_speed: 1.0
+min_speed: 0.0
+control: watermark
+
+[filament_switch_sensor filament_sensor]
+pause_on_runout: true
+switch_pin: PC1
+
+[probe]
+pin: PC14
+x_offset:45.2
+y_offset:11.6
+speed:5
+lift_speed:15
+z_offset = 2.350


Same as above

—
Reply to this email directly, view it on GitHub<https://github.com/Klipper3d/klipper/pull/6534#pullrequestreview-1964964249>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ARUZ2LMXZWVEC5NKLGQSUNDY2NJDBAVCNFSM6AAAAABE4MHAX6VHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMYTSNRUHE3DIMRUHE>.
You are receiving this because you authored the thread.Message ID: ***@***.***>

Sineos commented on test/klippy/printers.test:
```diff
@@ -203,6 +203,7 @@ CONFIG ../../config/printer-voxelab-aquila-2021.cfg
 
 # Printers using the stm32f401
 DICTIONARY stm32f401.dict
+CONFIG ../../config/printer-artillery-sidewinder-x3-plus-2024.cfg
 CONFIG ../../config/generic-fysetc-cheetah-v2.0.cfg
 CONFIG ../../config/printer-artillery-sidewinder-x2-2022.cfg
 CONFIG ../../config/printer-creality-ender5-s1-2023.cfg
```
Little nitpick: Please keep these lists sorted

```
CONFIG ../../config/generic-fysetc-cheetah-v2.0.cfg
CONFIG ../../config/printer-artillery-sidewinder-x2-2022.cfg
CONFIG ../../config/printer-artillery-sidewinder-x3-plus-2024.cfg
CONFIG ../../config/printer-creality-ender5-s1-2023.cfg
CONFIG ../../config/printer-elegoo-neptune3-pro-2023.cfg
```

TheFeralEngineer replied:
fixed 😋

Sineos reviewed (approved):
Thanks for your work. Appreciated. LGTM with a little nit.

TheFeralEngineer commented:
Yeah, sorry. Wasn't sure where to put it. I originally had it at the bottom
of the dictionary, but the example that was sent to me showed the most
recent addition at the top, that's why I put it there 🥴

Phil T.
The Feral Engineer

www.theferalengineer.com

Catnip contributions 🐱🐱
https://paypal.me/timpsonmusicllc


On Thu, Mar 28, 2024, 2:29 AM Sineos ***@***.***> wrote:

> ***@***.**** approved this pull request.
>
> Thanks for your work. Appreciated. LGTM with a little nit.
> ------------------------------
>
> In test/klippy/printers.test
> <https://github.com/Klipper3d/klipper/pull/6534#discussion_r1542379668>:
>
> > +CONFIG ../../config/printer-artillery-sidewinder-x3-plus-2024.cfg
>  CONFIG ../../config/generic-fysetc-cheetah-v2.0.cfg
>  CONFIG ../../config/printer-artillery-sidewinder-x2-2022.cfg
>  CONFIG ../../config/printer-creality-ender5-s1-2023.cfg
>
> Little nitpick: Please keep these lists sorted
>
> CONFIG ../../config/generic-fysetc-cheetah-v2.0.cfg
> CONFIG ../../config/printer-artillery-sidewinder-x2-2022.cfg
> CONFIG ../../config/printer-artillery-sidewinder-x3-plus-2024.cfg
> CONFIG ../../config/printer-creality-ender5-s1-2023.cfg
> CONFIG ../../config/printer-elegoo-neptune3-pro-2023.cfg
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Klipper3d/klipper/pull/6534#pullrequestreview-1965266877>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ARUZ2LOERGUIWFZAQZO5FFLY2OTDPAVCNFSM6AAAAABE4MHAX6VHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMYTSNRVGI3DMOBXG4>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

KevinOConnor commented:
Thanks.  So is this PR now ready to merge?

-Kevin

TheFeralEngineer commented:
it should be, yes. It passed all of the tests, I signed off on it, Sineos approved it but mentioned I should reorder the printers.test file, so I did. Other than that, everything should be all systems go.
________________________________
From: KevinOConnor ***@***.***>
Sent: Tuesday, April 2, 2024 9:51 PM
To: Klipper3d/klipper ***@***.***>
Cc: TheFeralEngineer ***@***.***>; Author ***@***.***>
Subject: Re: [Klipper3d/klipper] Update Kconfig (PR #6534)


Thanks. So is this PR now ready to merge?

-Kevin

—
Reply to this email directly, view it on GitHub<https://github.com/Klipper3d/klipper/pull/6534#issuecomment-2033394700>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ARUZ2LMXOATW6AZNQRO6RB3Y3NODZAVCNFSM6AAAAABE4MHAX6VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDAMZTGM4TINZQGA>.
You are receiving this because you authored the thread.Message ID: ***@***.***>

KevinOConnor commented:
Thanks.

-Kevin
